src/services/api/routes/communities.ts

```diff
@@ -22,27 +22,31 @@ import { ParamPack } from '@/types/common/param-pack';
 
 const createNewCommunitySchema = z.object({
 	name: z.string(),
-	description: z.string(),
+	description: z.string().optional(),
 	icon: z.string(),
-	app_data: z.string()
+	app_data: z.string().optional()
 });
 
 const router: express.Router = express.Router();
 
-function respondCommunityNotFound(response: express.Response) : void {
-	response.status(404);
-	response.send(xmlbuilder.create({
+function respondCommunityError(response: express.Response, http_code: number, error_code: number): void {
+	response.status(http_code).send(xmlbuilder.create({
 		result: {
 			has_error: 1,
 			version: 1,
-			code: 404,
-			error_code: 919,
-			message: 'COMMUNITY_NOT_FOUND'
+			code: http_code,
+			error_code: error_code,
+			message: 'COMMUNITY_ERROR' // This field is unused by the entire nn_olv.rpl
 		}
 	}).end({ pretty: true }));
 }
 
-async function commonGetSubCommunity(paramPack: ParamPack, communityID: string | undefined) : Promise<HydratedCommunityDocument | null> {
+function respondCommunityNotFound(response: express.Response): void {
+	respondCommunityError(response, 404, 919);
```

Comment:
Part of me almost wants to make an enum of all the supported HTTP status codes and Miiverse error codes that can be used here. Like `MiiverseError.CommunityNotFound` would be 919 and something like https://npmjs.com/package/http-status-codes for the HTTP status codes

Just so it's clear what these arguments are at a glance without having to look at the function signature? Not a huge deal either way, there's nothing "wrong" with the way it is now. Just some food for thought

Reply:
Agreed! I think Jemma had the same idea, it would make the code easier to understand for other people!

Reply:
Awesome! Do you wanna add this in to this PR or wait for a different one?

Reply:
I think it would best fit in a new PR, many if not all service files to change

Reply:
Sounds good! That's fair